packages/next/src/server/app-render/action-handler.ts

```diff
@@ -270,7 +305,7 @@ async function createRedirectRenderResult(
       process.env.__NEXT_PRIVATE_ORIGIN || `${proto}://${originalHost.value}`
 
     const fetchUrl = new URL(
-      `${origin}${basePath}${parsedRedirectUrl.pathname}${parsedRedirectUrl.search}`
+      `${origin}${appRelativeRedirectUrl.pathname}${appRelativeRedirectUrl.search}`
```

Comment:
`getAppRelativeRedirectUrl` is now responsible for including the `basePath` for both relative and absolute redirect urls.